pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/store/TextIndexUtils.java

```diff
@@ -36,6 +40,45 @@ static void cleanupTextIndex(File segDir, String column) {
   }
 
   static boolean hasTextIndex(File segDir, String column) {
-    return new File(segDir, column + V1Constants.Indexes.LUCENE_TEXT_INDEX_FILE_EXTENSION).exists();
+    return new File(segDir, column + V1Constants.Indexes.LUCENE_TEXT_INDEX_FILE_EXTENSION).exists() || new File(segDir,
+        column + V1Constants.Indexes.NATIVE_TEXT_INDEX_FILE_EXTENSION).exists();
+  }
+
+  public static boolean isFstTypeNative(@Nullable Map<String, String> textIndexProperties) {
+    if (textIndexProperties == null) {
+      return false;
+    }
+
+    for (Map.Entry<String, String> entry : textIndexProperties.entrySet()) {
+      if (entry.getKey().equalsIgnoreCase(FieldConfig.TEXT_FST_TYPE)) {
+        return entry.getValue().equalsIgnoreCase(FieldConfig.TEXT_NATIVE_FST_LITERAL);
+      }
+    }
+
+    return false;
+  }
+
+  public static FSTType getFSTTypeOfIndex(File segmentIndexDir, String columnName) {
+    boolean foundNativeTextIndex = false;
```

Comment:
This can be simplified to
```suggestion
    return SegmentDirectoryPaths.findTextIndexIndexFile(segmentIndexDir, columnName) != null ? FSTType.LUCENE : FSTType.NATIVE;
```

Note that we should not rely on whether the native text index file exists because for V3 format, there is no separate file for the native text index

Reply:
Lucene index is always stored as a separate file, but native index should be merged into the single index file for V3 format (not sure if we already do that)